include/util/static_graph.hpp

```diff
@@ -134,8 +134,11 @@ template <typename EdgeDataT, bool UseSharedMemory = false> class StaticGraph
     StaticGraph(typename ShM<NodeArrayEntry, UseSharedMemory>::vector &nodes,
                 typename ShM<EdgeArrayEntry, UseSharedMemory>::vector &edges)
     {
+        BOOST_ASSERT(!nodes.empty());
+
         number_of_nodes = static_cast<decltype(number_of_nodes)>(nodes.size() - 1);
-        number_of_edges = static_cast<decltype(number_of_edges)>(edges.size());
+        number_of_edges = static_cast<decltype(number_of_edges)>(nodes.back().first_edge);
+        BOOST_ASSERT(number_of_edges <= edges.size());
```

Comment:
The problem that the preallocated size in the memory block is `2 * #edges` to include duplicated edges, so shmem vector size is `2 * #edges`, but actual number of edges is the sentinel value after graph is loaded.  Some edges will be merged [here](https://github.com/Project-OSRM/osrm-backend/blob/ee076e61564a5770be6a518f1ecbe7c6daf67246/include/partition/edge_based_graph_reader.hpp#L154-L161)